packages/block-library/src/details/edit.js

```diff
@@ -42,6 +44,15 @@ function DetailsEdit( { attributes, setAttributes } ) {
 	const [ isOpen, setIsOpen ] = useState( showContent );
 	const dropdownMenuProps = useToolsPanelDropdownMenuProps();
 
+	// Check if the inner blocks are selected.
+	const hasSelectedInnerBlock = useSelect(
+		( select ) =>
+			select( blockEditorStore ).hasSelectedInnerBlock( clientId, true ),
+		[ clientId ]
+	);
+
+	const shouldRenderChildren = hasSelectedInnerBlock || isSelected || isOpen;
```

Comment:
This reintroduces a buggy behavior resolved via https://github.com/WordPress/gutenberg/pull/67766. We should avoid introducing regression while trying to solve other bugs.